src/main/groovy/JavadocGroupBuilder.groovy

```diff
@@ -0,0 +1,125 @@
+public class JavadocGroupBuilder {
+
+    private final String path
+    private final String artifactType
+    private final Set<String> artifactIDs
+    private final String pluginLocation
+
+    // AntBuilder used for unzipping content
+    private def indexHtml = new StringBuilder()
+    private def ant = new AntBuilder()
+    private def indexJson = new groovy.json.JsonBuilder()
+    private def jsonUrlMap = [:]
+
+
+    public JavadocGroupBuilder(String path, String artifactType, String title,
+                               Set<String> artifactIDs = null,
+                               String pluginLocation = "https://repo.jenkins-ci.org/releases/"
+                               ) {
+        this.path = path
+        this.artifactType = artifactType
+        this.artifactIDs = artifactIDs
+        this.pluginLocation = pluginLocation
+
+
+        // Header
+        indexHtml << "<html><head><title>${title}</title>"
```

Comment:
This code has been moved from the original impl. We could create an improvement request for that, but Javadoc repo is not so active to spend much time on that